miden-lib/asm/wallets/basic.masm

```diff
@@ -0,0 +1,28 @@
+use.miden::sat::account
+use.miden::sat::tx
+
+#! This procedure adds the given asset to the undelying account.
+#! 
+#! Inputs: [ASSET]
+#! Outputs: [0, 0, 0, 0, ...]
+#!
+#! - ASSET is the asset to be received, can be fungible or non-fungible
+export.receive_asset
+    exec.account::add_asset
+    dropw
+end
+
+#! Creates a note which sends a single asset to the specified recipient.
+#!
+#! Inputs: [ASSET, tag, RECIPIENT, ...]
+#! Outputs: [note_ptr, 0, 0, 0, 0, 0, 0, 0, 0, ...]
+#!
+#! - ASSET is the non-fungible asset of interest.
+#! - tag is the tag to be included in the note.
+#! - RECIPIENT is the recipient of the note.
+#! - note_ptr is the pointer to the memory address in the kernel.
+#!   This cannot directly be accessed from another context. 
+export.send_asset
+    exec.account::remove_asset
+    exec.tx::create_note
+end
```

Comment:
`tx::create_note` returns just the pointer (with all other inputs dropped from the stack). So, to pad the stack with 8 zeros after the pointer, we probably should do something like:
```
export.send_asset
    exec.account::remove_asset
    exec.tx::create_note
    padw padw movup.8
end
```

Reply:
Here the same thing happens, see comment above. Adding your suggestion results in 

```
ExecuteTransactionProgramFailed(InvalidStackDepthOnReturn(24))
```

So `send_assets` something seems off here.

Reply:
The situation here is somewhat similar to the one I described in https://github.com/0xPolygonMiden/miden-base/pull/233#discussion_r1320480096. But there is a complication: because `create_note` drops 8 elements from the stack, by the time we return from `create_note`, the stack looks something like this:

```
[note_ptr, x, x, x, x, x, x, x, 0, 0, 0, 0, 0, 0, 0, 0]
```

So, we can fix the problem by doing this:
```
export.send_asset
    exec.account::remove_asset
    exec.tx::create_note
    swapdw movup.8
end
```
There is one issue with the above though: we are assuming that `send_asset` will always be invoked via a `call` instruction. This should be the case most of the time, but it may also be invoked via an `exec` instruction if called from the account context. In such a case, the above code would result in an error.

So, I wonder if a better solution is to modify [tx::create_note](https://github.com/0xPolygonMiden/miden-base/blob/main/miden-lib/asm/sat/tx.masm#L68) to return `[ptr, 0, 0, 0, 0, 0, 0, 0, 0]` rather than just `[ptr]` (would actually make `tx::create_note` simpler. @frisitano - what do you think?

Also, noticed that there is a small documentation error [here](https://github.com/0xPolygonMiden/miden-base/blob/main/miden-lib/asm/sat/internal/tx.masm#L62) - the return type should be `[ptr, 0, 0, 0, 0, 0, 0, 0, 0]`, I believe.

Reply:
That looks about right. Maybe we create another issue on the above